### ISSUE TITLE
decorators with display: inline-block

### ISSUE THREAD

**Issue by mixtur** [facebook/draft-js#461]
When there are multiple lines in editor, arrow keys acts strangely in presence of decorators styled with "display: inline-block".

say "test" has decorator applied, `|`- is caret

xxxxxxxxxxxxx
xxx|xxx*test*xxxx
xxxxxxxxxxxxx

if you hit arrow down caret will go like this

xxxxxxxxxxxxx
xxxxxx|*test*xxxx
xxxxxxxxxxxxx

and any subsequent hits will not cause any caret movement

Another bad case is when caret is exactly after decorator

xxxxxxxxxxxxx
xxxxxx*test*|xxxx
xxxxxxxxxxxxx

then arrow up is not working

Here is the example code
https://gist.github.com/mixtur/c274147b4e3c9abe5dca370b8d96aca3

**Comment by hellendag** [comment]
Yeah, unfortunately I can't recommend using `display: inline-block` for decorators, for this reason. Cursor behavior can be really weird around these elements. It may be fixable with more investigation and testing, but mostly I've tried to avoid it.

Do you *need* to have an inline-block decorator for your use case?

**Comment by mixtur** [comment]
I am trying to implement emoji. 
I thought `display: inline-block` is natural for that purpose, since it is like inline but with manageable size. I'll try another approach. Monospaced fonts, I guess.

**Comment by ouchxp** [comment]
Run into the same issue, my solution is use `letter-spacing` to manage the size. Sadly I have to set different `letter-spacing` value for different browsers.
